### PR TITLE
Enhancement: Require localheinz/test-util

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] requires `localheinz/test-util`
* [x] uses `Localheinz\Test\Util\Helper` instead of `OpenCFP\Test\Helper\Faker\GeneratorTrait`
* [x] removes `OpenCFP\Test\Helper\Faker\GeneratorTrait`
* [x] uses assertions as provided by `Localheinz\Test\Util\Helper`

💁‍♂️ For reference, see [`localheinz/test-util`](https://github.com/localheinz/test-util).

### CODE REVIEW

File: tests/Unit/Domain/Services/AuthenticationExceptionTest.php [hunk 1]
```diff
@@ -21,24 +22,16 @@
...
-        $reflection = new \ReflectionClass(AuthenticationException::class);
-        $this->assertTrue($reflection->isFinal());
+        $this->assertClassIsFinal(AuthenticationException::class);
     }
-
-    public function testIsException()
-    {
-        $exception = new AuthenticationException();
-
-        $this->assertInstanceOf(\Exception::class, $exception);
```
Comment (localheinz) [inline tests/Unit/Domain/Services/AuthenticationExceptionTest.php]: Probably don't need to assert here that `RuntimeException` extends `Exception`.

**Comment by localheinz** [comment]
@BackEndTea @chartjes @mdwheele 

What do you think?

**Comment by BackEndTea** [comment]
👍 For me

**Comment by localheinz** [comment]
Thank you, @chartjes!

**Pull request merged**